behat/tests/bootstrap/v2/reporting/sections/VisitsCareSectionTrait.php

```diff
@@ -0,0 +1,304 @@
+
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @Given I choose no and save on the plans to move client to a new residence section
+     */
+    public function iChooseNoOnPlansToMoveClient()
+    {
+        $this->selectOption('visits_care[planMoveNewResidence]', 'no');
+        ++$this->answeredNo;
+
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @Given I choose yes and save on the plans to move client to a new residence section
+     */
+    public function iChooseYesOnPlansToMoveClient()
+    {
+        $this->selectOption('visits_care[planMoveNewResidence]', 'yes');
+        ++$this->answeredYes;
+
+        $info = 'The third set of information';
+        $this->fillField('visits_care[planMoveNewResidenceDetails]', $info);
+        array_push($this->additionalInfo, $info);
+
+        $this->pressButton('Save and continue');
+    }
+
+    /**
+     * @Then I should see the expected visits and care report section responses
+     */
+    public function iSeeExpectedVisitCareSectionResponses()
+    {
+        $table = $this->getSession()->getPage()->find('css', 'dl');
+
+        if (!$table) {
+            $this->throwContextualException('A dl element was not found on the page');
+        }
+
+        $tableEntry = $table->findAll('css', 'dd');
+
+        if (!$tableEntry) {
+            $this->throwContextualException('A dd element was not found on the page');
+        }
+
+        $countNegativeResponse = 0;
+        $countPositiveResponse = 0;
+
+        foreach ($tableEntry as $entry) {
+            if (1 == $this->careFundedChoice) {
+                assert(
+                    str_contains($entry, 'pays for all the care'),
+                    sprintf('matching care funding explanation %s ', 'Client pays for all the care')
+                );
+            } elseif (1 == $this->careFundedChoice) {
+                assert(
+                    str_contains($entry, 'gets some financial help'),
+                    sprintf('matching care funding explanation %s ', 'Client gets some financial help')
+                );
+            } elseif (1 == $this->careFundedChoice) {
+                assert(
+                    str_contains($entry, 'care is paid for by someone else'),
+                    sprintf('matching care funding explanation %s ', 'Client\'s care is paid for by someone else')
+                );
+            }
+
+            if ('no' === strtolower(trim($entry->getHtml()))) {
+                ++$countNegativeResponse;
+            } elseif ('yes' === strtolower(trim($entry->getHtml()))) {
+                ++$countPositiveResponse;
+            }
+        }
+
+        assert(
+            $countNegativeResponse == $this->answeredNo,
+            sprintf('Expected %d No responses, actual was %d', $this->answeredNo, $countNegativeResponse)
+        );
+
+        assert($countNegativeResponse == $this->answeredNo);
+        assert($countPositiveResponse == $this->answeredYes);
```

Comment:
I think @jamesrwarren created a wrapper around the `assert()` function that gives us a nice error output on what didn't match.

Jim - can you link to the code?

Reply:
https://github.com/ministryofjustice/opg-digideps/blob/main/behat/tests/bootstrap/v2/common/AssertTrait.php Yeah here you go. Choose between contains and equal wrappers based on what you're asserting and it makes the error a bit more consistent. There should be examples of where this is implemented around :)